utils/test-utils/README.md

```diff
@@ -0,0 +1,254 @@
+
+Inside this object you can find useful information about **Surface Duo 2** specs and utilities to span and unspan apps on this device and emulator.
+
+`fun switchFromSingleToDualScreen()`
+
+Span an app across displays. The function will handle correctly the different rotation modes of the device.
+This function works on **Surface Duo 2**.
+
+`fun switchFromDualToSingleScreen()`
+
+Unspan an app from being spanned across the whole display area to a single screen. The function will handle correctly the different rotation modes of the device.
+This function works on **Surface Duo 2**.
+
+### ViewMatcher
+
+Contains utilities that are used for view assertions in UI tests.
+
+```
+fun isViewOnScreen(
+    position: DisplayPosition,
+    orientation: Int,
+    firstDisplayWidth: Int,
+    totalDisplayWidth: Int,
+    foldingFeatureWidth: Int = 0
+): Matcher<View>
+```
+
+Returns a Matcher that checks if a View is shown in the display area given its position and screen dimensions.
+
+```
+fun areCoordinatesOnTargetScreen(
+    targetScreenPosition: DisplayPosition,
+    xStart: Int,
+    xEnd: Int,
+    firstDisplayWidth: Int,
+    totalDisplayWidth: Int,
+    foldingFeatureWidth: Int = 0
+): Boolean
+```
+
+Checks wether a specific screen-position is the screen defined by the function parameters.
+
+### Useful links
+
+- Documentation for these libraries at [dual-screen library](https://docs.microsoft.com/dual-screen/android/api-reference/dualscreen-library/)
+- Surface Duo developer documentation at [docs.microsoft.com/dual-screen](https://docs.microsoft.com/dual-screen).
+- [Surface Duo developer blog](https://devblogs.microsoft.com/surface-duo)
+
+### WindowLayoutInfoConsumer
+
+ Provides a [Consumer](https://developer.android.com/reference/java/util/function/Consumer)<[WindowLayoutInfo](https://developer.android.com/reference/kotlin/androidx/window/layout/WindowLayoutInfo)> that an Activity can use to reset previous WindowLayoutInfo data and to wait for new WindowLayoutInfo changes.
+
+`fun register(activity: AppCompatActivity)`
+
+Register a listener to consume WindowLayoutInfo values.
+
+`override fun accept(windowLayoutInfo: WindowLayoutInfo)`
+
+Accepts a WindowLayoutInfo with the new information that will handle.
+
+`fun resetWindowInfoLayout()`
+
+Resets the last window layout info to `null`.
+
+`fun resetWindowInfoLayoutCounter()`
+
+Resets WindowLayoutInfo counter when waiting for window layout changes to happen, before calling *waitForWindowInfoLayoutChanges*.
+
+`fun waitForWindowInfoLayoutChanges(): Boolean`
+
+Blocks and waits for the next window layout info changes to happen.
+Returns `true` if the window layout info changed before the timeout count reached zero, and `false` if the waiting time elapsed before the changes happened.
+
+`fun reset()`
+
+Resets current used instance to its original state.
+
+## Contributing
+
+Please read the [code of conduct](CODE_OF_CONDUCT.md) and [contribution guidelines](CONTRIBUTING.md).
```

Comment:
Both links lead to 404 errors.

Reply:
good catch! Thanks! Will change the `contributing` section entirely.

Reply:
Fixed